lib/pbench/server/database/models/datasets.py

```diff
@@ -1027,7 +1024,9 @@ def getvalue(dataset: Dataset, key: str, user_id: Optional[str] = None) -> JSON:
         else:
             try:
                 meta = Metadata.get(dataset, native_key, user_id)
-            except MetadataNotFound:
+            except Exception as e:
+                if not isinstance(e, MetadataNotFound):
+                    Metadata.logger.error("Unexpected exception {}", str(e))
```

Comment:
Is just logging it good enough?

I believe that the impetus for this change was that a Dashboard request was failing mysteriously, and there was little-to-no information about why.  This will no doubt help...once we conclude that the issue isn't on the Dashboard side and start looking on the Server side.  However, if we are getting an unexpected exception from `Metadata.get()` it seems to me that that **should** result in a `500` response to the client, which would prompt the developer to examine the Server log _first_....

So, rather than (or in addition to) logging here, it seems like we should be (re)raising an exception...no?

Ditto at line 1147.

Reply:
No, the dashboard problem was Nikhil's #3069. This is completely separate.

I was actually trying to make this a bit more robust by logging "unexpected" errors; and maybe I should have followed that by a raise.